velero/restore/restore.sh

```diff
@@ -0,0 +1,249 @@
+## TODO Check running pods in common-services namespace and wait for operand-deployment-lifecycle-manager pod to be running
+echo "You can watch the subscriptions come up in the Openshift Console. Go to Operators -> Installed Operators make sure the project selected is $CS_NAMESPACE."
+oc get pod -n $CS_NAMESPACE | grep operand-deployment-lifecycle-manager | grep Running
+while [ $? -ne 0 ]
+do
+  echo "Waiting for operand-deployment-lifecycle-manager pod to be running..."
+  oc get pod -n $CS_NAMESPACE
+  sleep 15
+  oc get pod -n $CS_NAMESPACE | grep operand-deployment-lifecycle-manager | grep Running
+done
+
+
+echo "Now that the subscriptions are in, we'll remove the default commonService object and restore our commonService object. Hit Enter:"
+read ANS
+
+# Restore the CommonService CR
+# A default is created during the subscription restore
+# We need to delete it, give it a chance to clear
+# then restore the one we have backed up
+oc delete commonservice common-service -n $CS_NAMESPACE
+echo "Waiting one minute for the deletion to complete..."
+sleep 30
+sed -i s/__BACKUP_NAME__/$BACKUP_NAME/g restore-commonservice.yaml
+oc apply -f restore-commonservice.yaml
+sleep 30
+sed -i s/$BACKUP_NAME/__BACKUP_NAME__/g restore-commonservice.yaml
+##Check status of common-service, should be Phase: Succeeded
+oc get commonservice common-service -n $CS_NAMESPACE
+
+# Restore licensing and cert-manager data
+# There are a number of configmaps, CRDs, Issuers, and Certificates that get restored here
+# Certificates usually take the longest, so we watch for those.
+echo "Now restoring licensing and cert-manager data. Hit Enter:"
+read ANS
+sed -i s/__BACKUP_NAME__/$BACKUP_NAME/g restore-licensing.yaml
+oc apply -f restore-licensing.yaml
+sleep 5
+sed -i s/$BACKUP_NAME/__BACKUP_NAME__/g restore-licensing.yaml
+
+sed -i s/__BACKUP_NAME__/$BACKUP_NAME/g restore-cert-manager.yaml
+oc apply -f restore-cert-manager.yaml
+sleep 5
+sed -i s/$BACKUP_NAME/__BACKUP_NAME__/g restore-cert-manager.yaml
+## Wait on certificates to populate
+oc get certificates -n $CS_NAMESPACE | grep cs-ca-certificate
+while [ $? -ne 0 ]
+do
+  echo "Waiting on certificates to be populated..."
+  sleep 10
+  oc get certificates -n $CS_NAMESPACE | grep cs-ca-certificate
+done
+
+echo "Now restoring the operand request. Hit Enter:"
+read ANS
+
+# Restore the operands
+# This is all the services within foundational service getting restored
+# We need MongoDB up so we can restore the mongodb data
+sed -i s/__BACKUP_NAME__/$BACKUP_NAME/g restore-operands.yaml
+oc apply -f restore-operands.yaml
+sleep 5
+sed -i s/$BACKUP_NAME/__BACKUP_NAME__/g restore-operands.yaml
+
+oc get pods -n $CS_NAMESPACE | grep icp-mongodb-0 | grep Running
+while [ $? -ne 0 ]
+do
+  echo "Waiting foundational services to be ready...Checking every minute"
+  oc get pods -n $CS_NAMESPACE
+  sleep 60
+  oc get pods -n $CS_NAMESPACE | grep icp-mongodb-0 | grep Running
+done
+echo "5 more minutes..."
+sleep 300
```

Comment:
A couple sleep statements that could be updated to look for something more specific to prevent possible race conditions.